scripts/functions.js

```diff
@@ -384,25 +431,29 @@ window.addEventListener('popstate', function() {
 	}
 })
 
-function zoomToPolygon(sourceID, coords, filterField) {
+function zoomToPolygon(sourceID, coords, filterField, maskLayer=true) {
 	if (typeof coords !== 'undefined') {
 		document.getElementById('statsBox').style.opacity = 0;
 		coords = coords.split(",");
 		bbox = [
 			[coords[0], coords[1]],
 			[coords[2], coords[3]]
 		];
-		updateURL(district=coords[4]);
-		if (coords[4] != '0') {
-			filterStates.district.val = coords[4];
-		}
-		if (showHouseDistricts) {
-			showHideLayer('state-house-districts-lines', markerNames=['state_house_districts'], showOnly=true);
-		} else if (showSenateDistricts) {
-			showHideLayer('state-senate-districts-lines', markerNames=['state_senate_districts'], showOnly=true);
+		if (maskLayer) {
+			updateURL(district=coords[4]);
+			if (coords[4] != '0') {
+				filterStates.district.val = coords[4];
+			}
+			if (showHouseDistricts) {
+				showHideLayer('state-house-districts-lines', markerNames=['state_house_districts'], showOnly=true);
+			} else if (showSenateDistricts) {
+				showHideLayer('state-senate-districts-lines', markerNames=['state_senate_districts'], showOnly=true);
+			} else if (showSchoolDistricts) {
+				showHideLayer('state-school-districts-lines', markerNames=['state_school_districts'], showOnly=true);
+			}
 		}
 		map.fitBounds(bbox, options={padding: 10, duration: 3000});
-		if (filterField !== undefined) {
+		if (maskLayer && filterField !== undefined) {
```

Comment:
This is the part where fading out unselected districts and filtering points are switched together.  If we decided to decouple them, I'll have to unpack this.